pkg/operation/care/constraints.go

```diff
@@ -139,6 +139,18 @@ func (c *Constraint) constraintsChecks(
 // CheckForProblematicWebhooks checks the Shoot for problematic webhooks which could prevent shoot worker nodes from
 // joining the cluster.
 func (c *Constraint) CheckForProblematicWebhooks(ctx context.Context) (gardencorev1beta1.ConditionStatus, string, string, error) {
+	// Earlier, the Gardener extensions library was not setting the timeout for auto-registered webhooks, hence, they
+	// ended up with the default (30s) which we consider problematic in shoots. For backwards-compatibility, we skip the
+	// check for shooted seeds to prevent it from failures caused by extension webhooks having a too high timeout value.
+	// See for more details: https://github.com/gardener/gardener/pull/3413#issuecomment-763974466
+	// TODO: Remove this in a future version.
+	if shootedSeed, err := gardencorev1beta1helper.ReadShootedSeed(c.shoot.Info); err == nil && shootedSeed != nil {
```

Comment:
Hm, I am confused here. Isn't `gardener-extension-provider-aws-shoot` installed in all Shoots? How the check for ShootedSeed helps here? I see this webhook created for all Shoots.

Reply:
```
error: shoots.core.gardener.cloud "aws-local" could not be patched: shoots.core.gardener.cloud "aws-local" is forbidden: MutatingWebhookConfiguration "gardener-extension-provider-aws-shoot" is problematic: webhook "shoot.aws.extensions.gardener.cloud" with failurePolicy "Ignore" and 30s timeout might prevent worker nodes from properly joining the shoot cluster
```

Reply:
Hm yeah, you're right, of course, I missed that. Shall we then simply skip the check for webhooks with the `gardener-extension-*-shoot` pattern? Not nice, but at least backwards-compatible and easy.

Reply:
Can't we set the timeout to `30s` for several releases to give time to extensions to adapt and after several releases we can set it back to 15s?

Reply:
We can, of course. But we can also simply add the name check and still keep @timebertt's improvement for end-user deployed webhooks, i.e., we get improved ops experience right with the next Gardener release.
My personal opinion is that setting it to `30s` and waiting a few releases would better if the "workaround" would be complex, but it isn't complex and shouldn't harm in any way.

Reply:
Okay, I am fine with both approaches then.

Reply:
I'm also fine with both approaches but slightly prefer the name check and exclusion of extension shoot webhooks to directly benefit from the enhanced checks with the next release.